R/exportDataQuality.R

```diff
@@ -0,0 +1,94 @@
+#####################################################################
+# exportDataQuality                                              ####
+
+#' @name exportDataQuality
+#' @title A helper function to export data queries from the Data Quality REDCap
+#'   module.
+#' @description Exports Data Quality queries by record. The Data Quality module 
+#'   must be enabled on the Control Center of REDCap to use this function. Additionally, 
+#'   this module must be enabled on each project before it can be used.
+#'
+#' @param rcon  A REDCap connection object as generated by `redcapConnection`.
+#' @param prefix A string from your REDCap institutions Data Quality module url. The 
+#'   module prefix can be found by exporting module settings under External Modules
+#'   in REDCap.
+  
+
+exportDataQuality <- function(rcon, prefix,
+                                     ...){
+  UseMethod("exportDataQuality")
+}
+
+exportDataQuality.redcapApiConnection <- function(rcon, prefix,
+                                                       ..., 
+                                                       error_handling = getOption("redcap_error_handling"), 
+                                                       config = list(), 
+                                                       api_param = list()){
+  ###################################################################
+  # Argument Validation                                          ####
+  
+  coll <- checkmate::makeAssertCollection()
+  
+  checkmate::assert_class(x = rcon, 
+                          classes = "redcapApiConnection", 
+                          add = coll)
+  
+  checkmate::assert_class(x = prefix, 
+                          classes = "character", 
+                          add = coll)
+  
+  error_handling <- checkmate::matchArg(x = error_handling, 
+                                        choices = c("null", "error"),
+                                        .var.name = "error_handling",
+                                        add = coll)
+  
+  checkmate::assert_list(x = config, 
+                         names = "named", 
+                         add = coll)
+  
+  checkmate::assert_list(x = api_param, 
+                         names = "named", 
+                         add = coll)
+  
+  checkmate::reportAssertions(coll)
+  
+  ###################################################################
+  # Build the query list                                         ####
+  
+  url <- paste0(rcon$url, "?prefix=", prefix, "&page=export&type=module&NOAUTH&pid=", rcon$projectInformation()$project_id)
+  
+  formData <- list(token = rcon$token)
+  
+  response <- httr::POST(url, body = formData, encode = "form")
+
+  tryCatch({
+    result <- httr::content(response, type = 'application/json')
+    
+    for(j in 1:length(result)){i=result[[j]];if(is.null(i$resolutions)){result[[j]]$resolutions=list()}}
```

Comment:
Use `seq_along(result)` instead of 1:length

Reply:
updated to use `seq_along(result)`